### DISCUSSION TITLE
[Proposal] Agile methods

### DISCUSSION THREAD

**Discussion by JustNrik** [dotnet/csharplang#2576] [Language Ideas]
Original proposal: https://github.com/dotnet/vblang/issues/115

Basically, a method with `agile` keyword (or whatever keyword fits) will implicitly have `.ConfigureAwait(false)` on each awaited expression. Useful for async libraries where you have to put it everywhere.

**Comment by scalablecory** [comment]
You may be able to use [this technique](https://blogs.msdn.microsoft.com/benwilli/2017/02/09/an-alternative-to-configureawaitfalse-everywhere/) instead of putting `ConfigureAwait` everywhere.

**Comment by bhood-zorus** [comment]
I'd like to toss in an opinion: It's unfortunate that `ConfigureAwait(true)` seems to be the unwanted case in pretty much everything but front-end code, but I don't think it would be appropriate to address it at the language level. Async function signatures are already unintuitive to learn, and this proposal is effectively leaking a .NET implementation detail into the language.

I'm under the impression that it's best practice to pretty much always use `ConfigureAwait(false)` in libraries, so my suggestion is that this be a compiler flag/something in the project properties. The language can go untouched, and the rare functions that need `ConfigureAwait(true)` can explicitly say so if the project is opted-in.

**Comment by ufcpp** [comment]
@brandonhood 
That's https://github.com/dotnet/csharplang/issues/2542.

**Comment by bhood-zorus** [comment]
@ufcpp Thanks for the link!
